core/src/main/java/com/linecorp/armeria/client/metric/MetricCollectingClient.java

```diff
@@ -50,10 +54,18 @@
     public static Function<? super HttpClient, MetricCollectingClient> newDecorator(
             MeterIdPrefixFunction meterIdPrefixFunction) {
         requireNonNull(meterIdPrefixFunction, "meterIdPrefixFunction");
-        return delegate -> new MetricCollectingClient(delegate, meterIdPrefixFunction);
+        return builder(meterIdPrefixFunction).newDecorator();
+    }
+
+    /**
+     * Returns a new {@link MetricCollectingClientBuilder} instance.
+     */
+    public static MetricCollectingClientBuilder builder(MeterIdPrefixFunction meterIdPrefixFunction) {
+        return new MetricCollectingClientBuilder(meterIdPrefixFunction);
     }
 
-    MetricCollectingClient(HttpClient delegate, MeterIdPrefixFunction meterIdPrefixFunction) {
-        super(delegate, meterIdPrefixFunction);
+    MetricCollectingClient(HttpClient delegate, MeterIdPrefixFunction meterIdPrefixFunction,
+                           @Nullable Predicate<? super RequestLog> successFunction) {
```

Comment:
How about using `BiPredicate`?
`BiPredicate<? super RequestContext, ? super RequestLog> successFunction`
so that a user can use the context to make a decision?